projects/driver_controls/src/stm32f0xx/ads1015.c

```diff
@@ -0,0 +1,201 @@
+#include "ads1015.h"
+#include <status.h>
+#include <stdio.h>
+#include "ads1015_def.h"
+#include "gpio_it.h"
+
+static void prv_check_status(StatusCode code) {
+  if (!code) {
+    status_msg(code, "failed");
+  }
+}
+
+// reset the internal registers and enter a power-down state
+static StatusCode prv_reset(ADS1015Data *data) {
+  uint8_t reset = ADS1015_RESET_BYTE;
+  StatusCode code = i2c_write(data->i2c_port, ADS1015_I2C_GENERAL_CALL, &reset, sizeof(reset));
+  prv_check_status(code);
+  status_ok_or_return(code);
+  return STATUS_CODE_OK;
+}
+
+// writes to register given upper and lower bytes
+static StatusCode prv_setup_register(ADS1015Data *data, uint8_t reg, uint8_t msb, uint8_t lsb) {
+  uint8_t ads1015_setup_register[] = { reg, msb, lsb };
+  StatusCode code = i2c_write(data->i2c_port, data->i2c_addr | ADS1015_I2C_ADDRESS_GND,
```

Comment:
In this case, since your address is offset from a base, you'll probably want to use `+`. This will still work because it's `0b1001000`, but imagine if your base address was actually `0b1001001`.